### ISSUE TITLE
Can I render a depth image with the renderer?

### ISSUE THREAD

**Issue by XLechter** [facebookresearch/pytorch3d#741]
Hi, thank you for your greate work first.
I wonder if/how I can render a depth image withdifferentiable rendering?
Thanks.

**Comment by patricklabatut** [comment]
Use `MeshRendererWithFragments`: its `forward()` method returns an image and the corresponding fragments. The `zbuf` property of the fragments can then be used to generate a depth image. See [the documentation of the mesh rasterizer](https://github.com/facebookresearch/pytorch3d/blob/master/pytorch3d/renderer/mesh/rasterize_meshes.py#L111-L116) for more details.

**Comment by luckybird1994** [comment]
Hi, I want to ask another problem. I wonder how I can render a depth image from pointcloud with differentiable rendering? Thanks.

**Comment by XLechter** [comment]
Hi,@patricklabatut , thank you for replying.
I tried the zbuf property you mentioned and it can create a depth image well. But when I tried to use the depth image for supervision to fit a mesh, the loss is exploded. The only modification is I change the silhouette image to depth image to compute loss. Could you give me some suggestions why the loss cannot converge?
Thank you so much!

**Comment by Wi-sc** [comment]
Hi, @XLechter. I'm also trying to find a way to generate depth maps. I think the reason why the mentioned way cannot supervise training is that only the closest faces are used. Of course if you have other ideas about it, please tell me and maybe we can talk about this problem.

**Comment by nikhilaravi** [comment]
@XLechter can you provide a small code snippet to show how you are initializing the renderer and how you are computing the loss?
